### PR TITLE
Network Controller refactor to use the same selector

### PR DESCRIPTION
**Pull request by tommasini**
**Description**
Network controller needs to be refactored for be easier to upgrade on the future.

The controller packages have been updated to use the latest versions of each package. The only breaking change that affects mobile was the rename of the provider property of the network controller to providerConfig. Most changes in this commit relate to that change.

**Proposed Solution**
Now we use the reselect package of redux to memoise and improve performance, and also we have a dedicated file to export the Network controller attributes.

**Test Cases**
* Overall use of the app (navigate between screens, change network, added and remove custom networks)
* Dapp Transactions on MetaMask test dapp
* Swap Ethereum to DAI on mainnet
* Send flow transaction of goerli
* Imported tokens on avalanche
* Imported nfts on mainnet

**Code impact**
High - Everything network related

**Screenshots/Recordings**
https://recordit.co/00LpJxmclu

**Issue**

Progresses #5509 

**Checklist**

* [x] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

**Review by Gudahtt — APPROVED** [review]
LGTM, great work!

**Review by blackdevelopa — APPROVED** [review]
This looks good to me. I can't think of network instance that was missed. Do you think we should run this by QA? What do you think?

**Comment by tommasini** [comment]
Thanks for the review!
Since there were major versions updates on some controllers package it would be good to do QA testing before merging it on my opinion

**Comment by Gudahtt** [comment]
Heads up @tommasini, it seems that there are more conflicts

File: app/components/Nav/Main/index.js [hunk 1]
```diff
@@ -218,9 +221,7 @@ const Main = (props) => {
   /**
    * Current network
    */
-  const networkProvider = useSelector(
-    (state) => state.engine.backgroundState.NetworkController.provider,
-  );
+  const networkProvider = useSelector((state) => selectProviderConfig(state));
```
Comment (Cal-L) [inline app/components/Nav/Main/index.js:224]: can simplify with `const networkProvider = useSelector(selectProviderConfig);`

File: app/components/UI/AccountRightButton/index.tsx [hunk 2]
```diff
@@ -67,8 +68,8 @@ const AccountRightButton = ({
   /**
    * Current network
    */
-  const networkProvider = useSelector(
-    (state: any) => state.engine.backgroundState.NetworkController.provider,
+  const networkProvider = useSelector((state: any) =>
```
Comment (Cal-L) [inline app/components/UI/AccountRightButton/index.tsx:71]: refer to simplify comment

File: app/components/UI/NetworkInfo/index.tsx [hunk 3]
```diff
@@ -112,31 +112,21 @@ interface NetworkInfoProps {
...
-    };
-    rpcTarget: string;
-    chainId: string;
-  };
   isTokenDetectionEnabled: boolean;
 }
 
 const NetworkInfo = (props: NetworkInfoProps) => {
+  const { onClose, ticker, isTokenDetectionEnabled } = props;
+  const networkProvider = useSelector((state: any) =>
```
Comment (Cal-L) [inline app/components/UI/NetworkInfo/index.tsx:120]: Simplify

File: app/components/Views/Wallet/index.tsx [hunk 4]
```diff
@@ -135,8 +138,8 @@ const Wallet = ({ navigation }: any) => {
   /**
    * Current network
    */
-  const networkProvider = useSelector(
-    (state: any) => state.engine.backgroundState.NetworkController.provider,
+  const networkProvider = useSelector((state: any) =>
```
Comment (Cal-L) [inline app/components/Views/Wallet/index.tsx:141]: Simplify

File: app/components/hooks/useAccounts/useAccounts.ts [hunk 5]
```diff
@@ -37,9 +41,7 @@ const useAccounts = ({
     (state: any) =>
       state.engine.backgroundState.PreferencesController.identities,
   );
-  const network = useSelector(
```
Comment (Cal-L) [inline app/components/hooks/useAccounts/useAccounts.ts:40]: simplify

File: app/components/hooks/useAccounts/useAccounts.ts [hunk 6]
```diff
@@ -57,10 +59,7 @@ const useAccounts = ({
     (state: any) =>
       state.engine.backgroundState.CurrencyRateController.currentCurrency,
   );
-  const ticker = useSelector(
-    (state: any) =>
-      state.engine.backgroundState.NetworkController.provider.ticker,
-  );
+  const ticker = useSelector((state: any) => selectTicker(state));
```
Comment (Cal-L) [inline app/components/hooks/useAccounts/useAccounts.ts:62]: simplify

**Review by Cal-L — COMMENTED** [review]
Left some comments! Mostly just simplifying the selectors

File: app/components/Nav/Main/index.js (hunk 1 above)
Comment (tommasini) [reply app/components/Nav/Main/index.js:224]: Doing this refactor, thanks for the review mate 🚀

**Review by cortisiko — APPROVED** [review]
I completed a thorough regression on this PR. This is QA passed! 🌮 🌮

**Pull request merged**